apps/server/src/shared/domain/entity/coursegroup.entity.ts

```diff
@@ -11,9 +13,13 @@ export interface ICourseGroupProperties {
 	students?: User[];
 }
 
+export interface ICourseGroupParent {
+	getUserIds(): EntityId[];
```

Comment:
i think the name `ICourseGroupParent` somehow does not link in my head to the userIds

Reply:
Yes and no. It has nothing todo with the getUserIds it self. It is a interface that must be fullfilled by any entity that come a parent of the coursegroup. It can be adding more methodes that must be required. If you look into task the parentData that are exposed dirctly for now should be also hide behind a method that is enforced over this way.

Reply:
Bei der Lesson würde ich das deutlich verständlicher finden: 
`export type ILessonParent = CourseGroup | Course;`

Reply:
So einen Type zu definieren wäre aber etwas anderes. Das Interface legt generell die Anforderungen fest (für beliebige Typen), der Type würde spezielle Typen erlauben.